utils/dask-setup.sh

```diff
@@ -24,75 +24,141 @@ echo -e "\n"
+                env CUDA_VISIBLE_DEVICES=$devs screen -dmS gpu_worker_$worker_id \
+                                                           bash -c 'script --flush --command "$create_worker" "$logfile"'
+            else
+                export create_worker="source activate $ENVNAME && \
+                                      dask-worker $MASTER_IPADDR:$DASK_SCHED_PORT \
+                                                  --host=${MY_IPADDR[0]} --no-nanny \
+                                                  --nprocs=1 --nthreads=1 \
+                                                  --memory-limit=0 --name ${MY_IPADDR[0]}_gpu_$worker_id \
+                                                  --local-directory $DASK_LOCAL_DIR/$name"
+                env CUDA_VISIBLE_DEVICES=$devs screen -dmS gpu_worker_$worker_id \
+                                                           bash -c "$create_worker"
+            fi
+            
+        done
     fi
-    echo "... starting gpu worker $worker_id"
-
-    if [[ "$DEBUG" = "DEBUG" ]]; then
-        export create_worker="source activate $ENVNAME && \
-                              cuda-memcheck dask-worker $MASTER_IPADDR:$DASK_SCHED_PORT \
-                                                        --host=${MY_IPADDR[0]} --no-nanny \
-                                                        --nprocs=1 --nthreads=1 \
-                                                        --memory-limit=0 --name ${MY_IPADDR[0]}_gpu_$worker_id \
-                                                        --local-directory $DASK_LOCAL_DIR/$name"
-        export logfile="${DASK_LOCAL_DIR}/gpu_worker_${worker_id}_log.txt"
-        env CUDA_VISIBLE_DEVICES=$devs screen -dmS gpu_worker_$worker_id \
-                                                   bash -c 'script -c "$create_worker" "$logfile"'
-    else
-        export create_worker="source activate $ENVNAME && \
-                              dask-worker $MASTER_IPADDR:$DASK_SCHED_PORT \
-                                          --host=${MY_IPADDR[0]} --no-nanny \
-                                          --nprocs=1 --nthreads=1 \
-                                          --memory-limit=0 --name ${MY_IPADDR[0]}_gpu_$worker_id \
-                                          --local-directory $DASK_LOCAL_DIR/$name"
-        env CUDA_VISIBLE_DEVICES=$devs screen -dmS gpu_worker_$worker_id \
-                                                   bash -c "$create_worker"
-    fi
-
-    WIDS[$id]=$!
-    done
-    sleep 5
-
-    echo -e "\n"
-
-    echo "... $NWORKERS worker(s) successfully started"
+fi
 
-    echo -e "\n"
+if [[ "$ARCH" = "CPU" ]]; then
+
+    if [[ "0" -le "$NWORKERS" ]]; then
+
+        if [[ "$WHOAMI" = "MASTER" ]]; then
+
+            echo "initializing dask scheduler..."
+            screen -dmS dask_scheduler bash -c "source activate $ENVNAME && dask-scheduler"
+            sleep 5
```

Comment:
Why is this sleep required?

Reply:
To ensure that the scheduler is launched before the workers.